### DISCUSSION TITLE
FreeDNS setup error for Nightscout on Google cloud

### DISCUSSION THREAD

**Discussion by Labonte19** [NightscoutFoundation/xDrip#2481] [Q&A]
I’m in the process of setting up Nightscout on Google Cloud. I’ve successfully installed Nightscout on VM but it keeps having internal errors when trying to setup FreeDNS. I’ve tried changing the subdomain but that doesn’t seem to help. I must of tried 10+ times now. Any suggestions?

**Comment by tzachi-dar** [comment]
Can you please add the file /var/log/letsencrypt/letsencrypt.log

Do you have a file called:  /etc/free-dns.sh
If you do, can you please do:
cat /etc/free-dns.sh 
It should show something like
#!/bin/sh
export HOSTNAME=xxxx.mooo.com
export DIRECTURL=https://freedns.afraid.org/dynamic/update.php?d005WVNjWkE3aDhyWkJpa1h1cFZiOUV5OjIwNzQxxxxx

Can you try running (replace xxx.mooo.com with your hostname)
sudo certbot '--nginx' '-d' 'xxxx.mooo.com' '--redirect' '--agree-tos' '--no-eff-email'

Please post here the results.
If the last command has passed successfully, this should mean that your nightscout site was installed correctly. Please try that it works.

**Reply by DomDevD** [reply]
Had the same issue as OP and this resolved it. Thanks!

**Comment by tzachi-dar** [comment]
@DomDevD @Labonte19 have you being prompted to enter API_SECREAT?
If not, than please let us know, and also go to the menu -> "Edit variables in a browser" and enter a new one.

**Reply by DomDevD** [reply]
In the beginning, no, I kept getting the freeDNS error.
Following the steps you mentioned, I then went back to step 2 of the installation phase and was finally asked to enter my API secret.
I was under the impression it worked but now I'm getting errors that my API secret is incorrect. However, when I check the variable value it is exactly what I'm entering.

**Comment by tzachi-dar** [comment]
After changing the api secret you need to reboot the server. Did you do it?

**Reply by DomDevD** [reply]
My bad. I thought I did but it's working now. Thanks again! :)
I'm now in the process of setting up a new profile.

**Comment by kilovictor29745** [comment]
I had the same issue with the FreeDNS error, as well as not getting an API Secret on step 2 of the installation.   After attempting step 2 2 or 3 times, I rebooted the VM and it worked.

**Comment by tzachi-dar** [comment]
For anyone getting here, rebooting the machine will solve the problem, and freedns will work after that.
